pkg/installmanager/installmanager.go

```diff
@@ -788,6 +842,49 @@ func (m *InstallManager) generateAssets(cd *hivev1.ClusterDeployment) error {
 	return nil
 }
 
+// patchWorkerMachineSetManifest accepts a yaml manifest as []byte and patches the manifest to include an additional
+// security group filter if that manifest is the definition of a worker MachineSet. The security group is obtained from
+// an annotation (constants.ExtraWorkerSecurityGroupAnnotation) on the provided MachinePool and is the value of the
+// annotation (singular).
+func patchWorkerMachineSetManifest(manifestBytes []byte, pool *hivev1.MachinePool, logger log.FieldLogger) (modifiedBytes []byte, modified bool, err error) {
+	c, err := yamlutils.Decode(manifestBytes)
+	// Decoding the yaml here shouldn't produce an error. An error indicates that the
+	// installer produced yaml that could not be decoded.
+	if err != nil {
+		logger.WithError(err).Error("unable to decode manifest bytes")
+		return modifiedBytes, modified, err
+	}
+
+	// Return if file is not a worker MachineSet manifest
+	isMachineSet, _ := yamlutils.Test(c, "/kind", "MachineSet")
+	// Note: We have to escape the `/` in the label key as `~1` per https://datatracker.ietf.org/doc/html/rfc6901#section-3
+	isWorkerMachineSet, _ := yamlutils.Test(c, "/spec/template/metadata/labels/machine.openshift.io~1cluster-api-machine-type", "worker")
```

Comment:
nit: it would be more efficient to short-circuit (bail if the first one fails, without checking the second).